app/src/main/java/org/sagebase/crf/step/CrfFitBitStepLayout.java

```diff
@@ -118,11 +128,28 @@ public void goForwardClicked(View v) {
 
     @Override
     protected void onComplete() {
-        if (fitbitManager.isAuthenticated()) {
+        if (fitbitManager.isAuthorized()) {
             super.onComplete();
         } else {
-            fitbitManager.authenticate();
+            Intent authIntent = fitbitManager.getAuthorizationIntent();
+            getActivity().startActivityForResult(authIntent, REQUEST_CODE);
+        }
+    }
+
+    private Activity getActivity() {
+        Context context = getContext();
```

Comment:
The easiest way to get the activity is to just check (callbacks instanceof Activity) which it always is, and use that.

Reply:
Thanks. This was a holdover from me trying to access the activity in initialize so I could schedule a call to onComplete in the next UI thread loop. The callback isn't set yet in when ViewTaskActivity calls initialize, and while I could've changed the order of calls, it seemed unwise to call onComplete before initialize (and super.initialize) completely finished. But it ended up being much simpler (for once) just to use RxJava to accomplish this